micro_mg3_0.F90

```diff
@@ -1022,7 +1045,74 @@ subroutine micro_mg_tend ( &
   deltat  = deltatin
   rdeltat = 1._r8 / deltat
 
+  if (trim(micro_mg_precip_frac_method) == 'in_cloud') then
+     precip_frac_method =  MG_PRECIP_FRAC_INCLOUD
+  else if(trim(micro_mg_precip_frac_method) == 'max_overlap') then
+     precip_frac_method = MG_PRECIP_FRAC_OVERLAP
+  endif
+
+  !===============================================
+  ! set mtime here to avoid answer-changing
+  mtime=deltat
+
+  !......................................................................
+  !       graupel/hail density set (Hail = 400, Graupel = 500 from M2005)
+  bgtmp=0._r8
+  rhogtmp=0._r8
+  if (do_hail) then
+     bgtmp = bh
+     rhogtmp = rhoh
+  end if
+  if (do_graupel) then
+     bgtmp = bg
+     rhogtmp = rhog
+  end if
+
+  mdust = size(rndst,3)
```

Comment:
Should add a comment above this line that says "set mdust as the number of dust bins for use later in contact freezing subroutine"

Reply:
Thanks Kate. Just add it.